### PR TITLE
feat(desktop): Qraft token 文件通道 + 生产环境 client_secret 默认值（#726 跟进）

### PR DESCRIPTION
**Pull request by 14790897**
## 类型

- [x] ✨ 新功能

## 变更概述

Qraft 登录态 token 文件通道落地（供 Skill/agent 读取 access_token）+ 生产环境 client_secret 增加测试阶段默认值。

## 背景/问题

#726（PR #728）内置 Qraft OAuth2 登录后，access_token 只存在于主进程内存与 safeStorage 加密文件中，agent（Python 后端）无法获取。#674 方案上传 Skill 的凭据方案需要一条「从 MiQi 获取 access_token 的通道」作为后续收敛目标（详见 docs/frontend/qraft-oauth2-login.md 第 6 节）。同时用户要求生产环境 client_secret 在测试阶段同样开箱即用。

## 关联 issue

- #726 的跟进（token 通道是 #674 凭据方案收敛的前提）

## 变更内容

- `apps/desktop/src/main/qraft/service.ts`：QraftServiceOptions 新增 `tokenFilePath` 注入；登录/刷新成功后写 `<workspace>/.qraft/token.json`（0600，仅含 accessToken + expiresAt，不含 refresh_token），退出登录删除，应用启动恢复登录态时同步重写
- `apps/desktop/src/main/qraft/ipc.ts`：按 config 解析 workspace 作为 token 文件位置（workspace 在 KUN 沙箱中 bind-mount 到 /home/miqi/workspace，沙箱内 Skill 可直接读取）；`apps/desktop/src/main/ipc/index.ts` 导出 getWorkspacePath 供复用
- `apps/desktop/src/main/qraft/types.ts`：生产环境 client_secret 硬编码默认值（测试阶段开箱即用），`QRAFT_PROD_CLIENT_SECRET` 环境变量可覆盖；生产环境仍强制注册 redirect_uri（validateConfig 不变）
- `apps/desktop/src/renderer/features/settings/components/QraftPage.tsx`：client_secret 输入框占位文案更新（留空用默认值）
- `docs/frontend/qraft-oauth2-login.md`：第 6 节 token 通道由设计建议改为已实现落地说明（文件格式/沙箱可达性/auth.py 读取策略/实施顺序）
- 测试：token 文件生命周期单测 4 例（登录写入/自动刷新更新/登出删除/tokenFilePath 为 null 时静默跳过）+ 生产 secret 默认值与覆盖单测；离线 Electron E2E 增加 token 文件存在性、不含 refresh_token、登出删除断言

## 日志/验证证据

```
$ npx vitest run
Test Files  18 passed | 1 skipped (19)
     Tests  256 passed | 2 skipped (258)

$ npm run typecheck && npm run build
（通过）

$ PLAYWRIGHT_SKIP_WEB_SERVER=1 npx playwright test tests/e2e/qraft-login.spec.ts --project=electron
3 passed（含 token 文件存在性/登出删除断言）
```

## 测试情况

- 全量单测 256 例通过（新增 token 文件通道 4 例、生产 secret 2 例）
- typecheck（node/web）+ electron-vite build 通过
- 离线 Electron E2E 3 例通过：预置登录态启动后 token 文件出现在临时 workspace/.qraft/token.json（含 e2e accessToken、无 refresh_token），退出登录后文件删除
- 真实环境浏览器登录 E2E 与本 PR 无关，未重跑

## 截图

无需截图（无 UI 视觉变更，仅占位文案微调）

## 后续计划

- #674 的 auth.py 按文档第 6 节读取策略落地：优先读 token 文件，自管凭据降级为兜底，稳定后收敛为纯「取 token + 过期检测」
- #749：token 文件「宿主目录存储 + 沙箱只读挂载」隔离加固（CodeRabbit 评审提出的完整隔离方案，本 PR 维持 workspace 内 lstat 防护 + 登出代际校验）
- 转正式环境接入前移除测试/生产 client_secret 硬编码默认值（types.ts 已标注）

> ⚠️ 叠放 PR：base 为 `claude/elastic-dewdney-fbf858`（PR #728 分支）。#728 合入 develop 后需把本 PR 的 base 改回 develop。

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Add token file at `workspace/.qraft/token.json` for Skill/agent access

- Introduce production `client_secret` default with `QRAFT_PROD_CLIENT_SECRET` override

- Harden token writes via 0600, symlink checks, logout generation guard

- Expand unit and E2E tests for token lifecycle and agent read path

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Qraft IPC"] --> B["QraftService with tokenFilePath"]
  B --> C["syncTokenFile"]
  C --> D["workspace/.qraft/token.json"]
  B --> E["deleteTokenFile on logout"]
  B --> F["prod client_secret default"]
  G["Unit/E2E tests"] --> B
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td><strong>index.ts</strong><dd><code>Export workspace path resolver for Qraft token file</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/14790897/MiQi/pull/747/files#diff-8884758a148ac81f6980064c69650ae427e276cf7752f4481578bd12abbc4258">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ipc.ts</strong><dd><code>Inject workspace token file path into QraftService</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/14790897/MiQi/pull/747/files#diff-5507eb6c5c1ad3762cc5913543210b872adf029f2f98ebddeb72ce3dd0088f59">+10/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>QraftPage.tsx</strong><dd><code>Update client_secret placeholder for default value</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/14790897/MiQi/pull/747/files#diff-df6f3faee730d7190408e2a46578ea94c0215bdfb3aba09fe2c42093bc920f74">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Security</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>service.ts</strong><dd><code>Persist tokens to file with logout and symlink protections</code></dd></td>
  <td><a href="https://github.com/14790897/MiQi/pull/747/files#diff-373c7a2a36fd357a77dca028fd31fd1bdc60e42c59c45e23392e6626d0a00674">+90/-5</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>types.ts</strong><dd><code>Add production client_secret default and override helper</code>&nbsp; </dd></td>
  <td><a href="https://github.com/14790897/MiQi/pull/747/files#diff-6741723697a09c9160fa8c299e9820f8959134f7241e6ffed2ccd1415e67e308">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>service.test.ts</strong><dd><code>Test token file lifecycle and edge cases</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/14790897/MiQi/pull/747/files#diff-08d39ead05325dcd6b17e8eea84d5f3b6d5244ac753a576781b483cca2f8e480">+157/-13</a></td>

</tr>

<tr>
  <td><strong>qraft-login.spec.ts</strong><dd><code>Verify token file and agent read path in E2E</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/14790897/MiQi/pull/747/files#diff-e6ac530d70356248cb5651eacf1ff07343e34779957831810597e4f8b5e7a18f">+27/-1</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>qraft-oauth2-login.md</strong><dd><code>Document token file channel implementation and production secret</code></dd></td>
  <td><a href="https://github.com/14790897/MiQi/pull/747/files#diff-4d367dba09018850dc76766d9ac85eab14ab49ebcfa885f64a001bd6cb2abd2e">+24/-23</a>&nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by 14790897** [comment]
回复两条自动化评审意见（PR Code Suggestions / PR Reviewer Guide）：

1. **「Remove hardcoded production client secret default」（Security，重要性 8）—— 不采纳**：保留 `miqi123456` 生产默认值是仓库维护者（issue 作者）的明确决策——当前处于测试阶段，开箱即用优先，且该值即测试阶段统一使用的临时凭据。已做的防护：`QRAFT_PROD_CLIENT_SECRET` 环境变量可覆盖；`types.ts` 注释标注「转正式环境接入前应移除默认值」；生产环境仍强制填写注册的 redirect_uri（validateConfig 拒绝缺失），并非完全放开。转正式接入时按该建议改回空字符串 + INVALID_CONFIG 拒绝即可（单行改动）。

2. **#674 部分合规（Skill 侧 auth.py 读取/upload_run.py/validation_report/确认卡片/上传重试未实现）—— 符合预期，不在本 PR 范围**：本 PR 只交付 #674 收敛方案的 Desktop 侧一半（token 文件通道），Skill 侧各条目由 #674 本体（已指派）承接，读取策略见 docs/frontend/qraft-oauth2-login.md 第 6 节。

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

**Comment by 14790897** [comment]
回复 PR Reviewer Guide（ticket compliance 分析）：

1. **#726 / #674 标记 "Not compliant" 是叠放 PR 的分析误报**：本 PR #747 的 base 是 `claude/elastic-dewdney-fbf858`（PR #728），合规分析只看到了两分支间的增量（token 文件通道 + 生产 secret 默认值）。#726 的全部需求（设置页 RSA 密码登录、授权码流程、token 刷新、错误分类）已在 base PR #728 交付并测试；#674 的 Skill 侧条目（auth.py/upload_run.py/validation_report/确认卡片）由 #674 本体承接，本 PR 只交付其 Desktop 侧前置（token 通道），已在 PR 描述标注。

2. **"plaintext token file instead of secure storage" 是误解**：secure storage（safeStorage 加密）不变，token 文件是给 agent/Skill 的**附加读取通道**，仅含 accessToken + expiresAt（0600、2 小时 TTL、登出删除），设计权衡见 docs/frontend/qraft-oauth2-login.md 第 6 节，两者并存互补而非替代。

3. **Hardcoded client secret**：维护者明确决策（测试阶段开箱即用），环境变量可覆盖、types.ts 已标注转正式接入前移除，上一条回复已详述。

无需代码变更。

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: apps/desktop/src/main/qraft/service.ts [hunk 1]
```diff
@@ -267,6 +279,45 @@ export class QraftService {
...
+      mkdirSync(dirname(filePath), { recursive: true });
+      writeFileSync(
+        filePath,
+        JSON.stringify({
+          accessToken: state.tokens.accessToken,
+          expiresAt: state.tokens.expiresAt,
+        }),
+        { encoding: 'utf8', mode: 0o600 }
+      );
+      chmodSync(filePath, 0o600);
```
[the remark this answers is not in the document]
Comment (14790897) [reply apps/desktop/src/main/qraft/service.ts]: 已修复（commit 29ec9d24）：
1. **登出竞态**：新增 authGeneration 登录代际 —— logout 递增并清空 inFlightRefresh；runRefresh 在 refreshTokens 返回后校验代际，不匹配则丢弃结果（不落盘、不写 token 文件、不重新调度）；tickRefresh 对已登出状态直接丢弃。单测覆盖「在途刷新完成于登出后 → store 与 token 文件均不恢复」。
2. **symlink 重定向**：写入前 lstat 校验 .qraft 为真实目录、既有文件为常规文件（非 symlink），否则跳过写入并 WARN；单测覆盖非目录占位（跨平台）与 symlink 拒绝（Windows 跳过）。

关于「移到宿主目录 + 只读挂载」的重架构建议：workspace bind-mount 是 agent 读取该文件的既有通道，本 PR 采用「写入前校验 + 登出代际」的轻量防护，保持零沙箱改动；如后续要求更强隔离（宿主目录 + ro-bind），可作为独立改造另行立项。

全量 258 单测 + typecheck + build 通过。

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

[the remark this answers is not in the document]
Comment (14790897) [reply apps/desktop/src/main/qraft/service.ts]: 同意，剩余 TOCTOU 风险已知并接受为测试阶段残余（攻击面即 agent 本身，agent 本就对 token 有读权限；lstat 校验防的是静态 symlink 替换与意外重定向）。「宿主目录存储 + 沙箱只读挂载」隔离改造已按建议立为后续 issue 跟踪：#749（Qraft token 文件宿主目录隔离 + 沙箱只读挂载），其中也记录了 bridge 反向调用作为更长期目标。

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

[the remark this answers is not in the document]
Comment (14790897) [reply apps/desktop/src/main/qraft/service.ts]: 确认：剩余 TOCTOU 风险由 #749 跟踪（宿主目录存储 + 沙箱只读挂载），本 PR 不再重复处理。本线程无待办，可以收起。

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

**Comment by 14790897** [comment]
@coderabbitai resolve

本 PR 的全部评审线程已处理完毕：登出竞态（authGeneration 代际校验）与 symlink 静态防护（lstat 校验）已修复并单测覆盖；剩余 TOCTOU 风险已立 #749 跟踪，不在本 PR 范围。
